src/components/common/DataElementSelector/DataElementSelector.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { Select } from "@madie/madie-design-system/dist/react";
+import { useQdmPatient } from "../../../util/QdmPatientContext";
+import { filterDataElements } from "../../../util/DataElementHelper";
+import { MenuItem } from "@mui/material";
+
+interface DataElementSelectorProps {
+  canEdit: boolean;
+  value?: string;
+  handleChange?: (value) => void;
+}
+interface MenuObj {
+  value: string;
+  label: string;
+}
+
+const DataElementSelector = ({
+  canEdit,
+  handleChange,
+  value,
+}: DataElementSelectorProps) => {
+  const { state } = useQdmPatient();
+  const { patient } = state;
+  const dataElements = filterDataElements(patient?.dataElements);
+  const options: MenuObj[] = dataElements
+    ? dataElements.map(({ id, description }) => ({
+        value: id,
+        label: description,
+      }))
+    : [];
```

Comment:
I think with this, you will be able to relate the current element with itself a kind of self-association. I don't think that's necessary. we could remove the current element.

Reply:
This prevents crashes during testing. This is not an issue during the normal flow of the application otherwise